File: java_from_ruby/overload_demo.rb
require 'java'
java_import 'OverloadDemo'

# START:overload
OverloadDemo.what_type_is 42       # => "long"
OverloadDemo.what_type_is "Fun!"   # => "string"
OverloadDemo.what_type_is Hash.new # => "object"
# END:overload

File: java_from_ruby/OverloadDemo.java
import java.util.List;

public class OverloadDemo {
    public static String whatTypeIs(long value) {
        return "long";
    }

    public static String whatTypeIs(String value) {
        return "string";
    }

    public static String whatTypeIs(Object value) {
        return "object";
    }
}
